crates/aptos-dkg/src/pvss/fiat_shamir.rs

```diff
@@ -61,85 +128,76 @@ impl<T: Transcript> FiatShamirProtocol<T> for merlin::Transcript {
+    fn append_auxs<A: Serialize>(&mut self, auxs: &[A]) {
         self.append_u64(b"auxs", auxs.len() as u64);
-
         for aux in auxs {
-            let aux_bytes = bcs::to_bytes(aux).expect("aux data serialization should succeed");
-            self.append_message(b"aux", aux_bytes.as_slice())
+            <merlin::Transcript as PVSS<S, T>>::append_aux::<A>(self, aux);
         }
     }
 
+    fn append_aux<A: Serialize>(&mut self, aux: &A) {
+        let aux_bytes = bcs::to_bytes(aux).expect("aux data serialization should succeed");
+        self.append_message(b"aux", aux_bytes.as_slice());
+    }
+
+    // fn append_share_commitments<A: Serialize>(&mut self, commitments: &[A]) {
+    //     let commitments_bytes = bcs::to_bytes(commitments).expect("share commitment serialization should succeed");
+    //     self.append_message(b"share_commitments", commitments_bytes.as_slice());
+    // }
+
     fn append_transcript(&mut self, trx: &T) {
         self.append_message(b"transcript", trx.to_bytes().as_slice());
     }
 
-    fn challenge_dual_code_word_polynomial(&mut self, t: usize, n_plus_1: usize) -> Vec<Scalar> {
+    fn challenge_dual_code_word_polynomial(&mut self, t: usize, n_plus_1: usize) -> Vec<S> {
         let num_coeffs = n_plus_1 - t;
-
-        let num_bytes = num_coeffs * 2 * SCALAR_NUM_BYTES;
-        let mut buf = vec![0u8; num_bytes];
-
-        self.challenge_bytes(b"challenge_dual_code_word_polynomial", &mut buf);
-
-        let mut f = Vec::with_capacity(num_coeffs);
-
-        for chunk in buf.chunks(2 * SCALAR_NUM_BYTES) {
-            match chunk.try_into() {
-                Ok(chunk) => {
-                    f.push(random_scalar_from_uniform_bytes(chunk));
-                },
-                Err(_) => panic!("Expected a slice of size 64, but got a different size"),
-            }
-        }
-
-        assert_eq!(f.len(), num_coeffs);
-
-        f
+        <merlin::Transcript as ScalarProtocol<S>>::challenge_scalars(
+            self,
+            b"challenge_dual_code_word_polynomial",
+            num_coeffs,
+        )
     }
 
-    fn challenge_linear_combination_scalars(&mut self, num_scalars: usize) -> Vec<Scalar> {
-        let mut buf = vec![0u8; num_scalars * 2 * SCALAR_NUM_BYTES];
-        self.challenge_bytes(b"challenge_linear_combination", &mut buf);
-
-        let mut v = Vec::with_capacity(num_scalars);
-
-        // To ensure we pick a uniform Scalar, we sample twice the number of bytes in a scalar and
-        // reduce those bytes modulo the order of the scalar field.
-        for chunk in buf.chunks(2 * SCALAR_NUM_BYTES) {
-            match chunk.try_into() {
-                Ok(chunk) => {
-                    v.push(random_scalar_from_uniform_bytes(chunk));
-                },
-                Err(_) => panic!("Expected a 64-byte slice, but got a different size"),
-            }
-        }
+    fn challenge_linear_combination_scalars(&mut self, num_scalars: usize) -> Vec<S> {
```

Comment:
I think you may not need this function anymore: just let the caller call `challenge_scalars` directly.